test/controllers/tracks_controller_test.rb

```diff
@@ -303,18 +304,14 @@ class TracksControllerAudioTest < ActionDispatch::IntegrationTest
     flac = Codec.create(mimetype: 'audio/flac', extension: 'flac')
     audio_file = create(:audio_file, location:, filename: '/base.flac', codec: flac)
     track = create(:track, audio_file:)
-    get audio_track_url(track, codec_conversion_id: codec_conversion.id)
-    File.delete(TranscodedItem.last.path)
+    transcode = TranscodedItem.create(audio_file:, codec_conversion:)
```

Comment:
This test failed before, because of the rather cryptic `IOError: not opened for reading` exception.

Creating the transcode without going through a request fixed this issue. IMO this also makes the intention of the test easier to read/understand.

Backtrace:
```
IOError: not opened for reading
    app/models/transcoded_item.rb:33:in `read'
    app/models/transcoded_item.rb:33:in `readpartial'
    app/models/transcoded_item.rb:33:in `copy_stream'
    app/models/transcoded_item.rb:33:in `block in do_conversion'
    app/models/transcoded_item.rb:33:in `open'
    app/models/transcoded_item.rb:33:in `do_conversion'
 ```

Reply:
Seeing as this is part of the `tracks_controller_test`, I don't entirely agree with not going through a request.

Reply:
The actual request that we want to test is on L313. This request was in no way checked or asserted, so I assumed this was part of the setup.
You wrote these tests, so let me know if I interpreted that wrong

Reply:
Ah, you're right. Hadn't looked at it in detail, my bad.